src/main/java/org/apache/commons/text/TextStringBuilder.java

```diff
@@ -1850,7 +1850,14 @@ public boolean equals(final Object obj) {
      * @return true if the builders contain the same characters in the same order
      */
     public boolean equals(final TextStringBuilder other) {
-        return other != null && Arrays.equals(buffer, other.buffer);
+    	
+        if(other == null) {
+            return false;
+        }
+        if (this.size != other.size) {
+            return false;
+        }
+        return this.toString().equals(other.toString());
```

Comment:
@sebx59 your change makes the code look very similar to the code prior to the change that added `Arrays.equals` (which I also wouldn't have guessed to cause a regression here).

Maybe we could just revert it instead? https://github.com/apache/commons-text/commit/36af8b1884a3ac7bf323a08a7d62e4a290336d2b

And also add a comment in the code so we don't accidentally add the regression again in the code. WDYT?

Reply:
done ! thanks